### PR TITLE
#258 Start in container option not appearing if parent folder of Liberty project is opened in VS Code

### PR DESCRIPTION
**Pull request by aparnamichael**
Added start in container option, even if the parent project is opened.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -176,7 +176,7 @@
 				},
 				{
 					"command": "liberty.dev.start.container",
-					"when": "viewItem == libertyMavenProjectContainer || viewItem == libertyGradleProjectContainer",
+					"when": "viewItem == libertyMavenProject || viewItem == libertyGradleProject || viewItem == libertyMavenProjectContainer || viewItem == libertyGradleProjectContainer",
```
Comment (TrevCraw) [inline package.json:179]: `libertyMavenProjectContainer` & `libertyGradleProjectContainer` are specific types of Liberty Maven and Liberty Gradle projects. We actually do not want the "start in container" option to show for any Liberty Maven or Liberty Gradle project, so unfortunately this solution is not an option.

File: src/liberty/devCommands.ts [hunk 2]
```diff
@@ -441,11 +441,11 @@ export async function startContainerDevMode(libProject?: LibertyProject | undefi
         if (terminal !== undefined) {
             terminal.show();
             libProject.setTerminal(terminal);
-            if (libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER) {
+            if (libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER || libProject.getContextValue() === LIBERTY_MAVEN_PROJECT ) {
```
Comment (TrevCraw) [inline src/liberty/devCommands.ts:444]: only LIBERTY_MAVEN_PROJECT_CONTAINER is valid here

File: src/liberty/devCommands.ts [hunk 3]
```diff
@@ -441,11 +441,11 @@ export async function startContainerDevMode(libProject?: LibertyProject | undefi
...
                 const mvnCmdStart = await mvnCmd(libProject.getPath());
                 const cmd = `${mvnCmdStart} io.openliberty.tools:liberty-maven-plugin:devc -f "${libProject.getPath()}"`;
                 terminal.sendText(cmd);
-            } else if (libProject.getContextValue() === LIBERTY_GRADLE_PROJECT_CONTAINER) {
+            } else if (libProject.getContextValue() === LIBERTY_GRADLE_PROJECT_CONTAINER || libProject.getContextValue() === LIBERTY_GRADLE_PROJECT) {
```
Comment (TrevCraw) [inline src/liberty/devCommands.ts:448]: only LIBERTY_GRADLE_PROJECT_CONTAINER is valid here

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/liberty/devCommands.ts (hunk 2 above)
Comment (aparnamichael) [reply src/liberty/devCommands.ts]: Reverted the changes

File: src/liberty/devCommands.ts (hunk 3 above)
Comment (aparnamichael) [reply src/liberty/devCommands.ts]: Reverted the changes

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/liberty/libertyProject.ts [hunk 4]
```diff
@@ -397,22 +397,22 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
...
+					newProjectsMap.set(pomFile, project);
 				}
 			} else {
 				const gradleFile = vscodePath.resolve(folder, "build.gradle");
-				if ( fse.existsSync (gradleFile) ) {
-					if ( !this.addExistingProjectToNewProjectsMap(gradleFile, LIBERTY_GRADLE_PROJECT, newProjectsMap) ) {
-						const project = await createProject(this._context, gradleFile, LIBERTY_GRADLE_PROJECT);
+				if (fse.existsSync(gradleFile) && !newProjectsMap.has(gradleFile)) {
+					if (!this.addExistingProjectToNewProjectsMap(gradleFile, LIBERTY_GRADLE_PROJECT, newProjectsMap)) {
+						const project = await createProject(this._context, gradleFile, LIBERTY_MAVEN_PROJECT);
```
[the remark this answers is not in the document]
Comment (aparnamichael) [reply src/liberty/libertyProject.ts]: @TrevCraw: made it as gradle project.

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**